stdlib/Test/src/Test.jl

```diff
@@ -1717,9 +1752,11 @@ function testset_beginend_call(args, tests, source)
         # by wrapping the body in a function
         local default_rng_orig = copy(default_rng())
         local tls_seed_orig = copy(Random.get_tls_seed())
+        local tls_seed = isnothing(get_seed(ts)) ? set_seed!(ts, tls_seed_orig) : get_seed(ts)
         try
             # default RNG is reset to its state from last `seed!()` to ease reproduce a failed test
-            copy!(Random.default_rng(), tls_seed_orig)
+            copy!(Random.default_rng(), tls_seed)
```

Comment:
It should be documented that the seed needs to be a copyable RNG, not something like `Random.RandomDevice` which can't be copied. Unfortunately, the only reliable way check for that is calling `copy` and seeing if it throws...

Reply:
Done, but I the way I see this option to be used is to reproduce failures observed on CI.  If someone wants to use a fancy RNG to make tests reproducible they should call `rand(rng)` anyway.

Reply:
100% agreed in principle, but unfortunately not every package takes an RNG object as part of their API, even if they internally call `rand`. I've hit that exact "bug" in Supposition.jl too, which is exactly why I now seed the default RNG from the one I provide :) 

One case that this still doesn't (and can't) cover is when a library doesn't use the default RNG at all, but rather some other source of randomness (e.g. as happened [here](https://github.com/Seelengrab/Supposition.jl/issues/56#issuecomment-2530505150)). [Arpack](https://arpack.julialinearalgebra.org/stable/) for example uses  an RNG internally, but that one doesn't seem to be seedable from Julia, so even with the feature from this PR, failures involving this library are going to be non-deterministic.